TgeeT/Questioner
# #163215141 fix signup validation and edge case bug

#### What does this PR do?

This PR is to ensure that all user data is validated and basic edge cases handled
#### Description of Task to be completed?

Write a base validation file called schema.py
Ensure the validation is done in the models and views of the user

#### How should this be manually tested?

Follow the steps below:
1. git clone https://github.com/tgeet/Questioner.git
2. git checkout to bg-signup-fix-163215141 branch
3. run the app and use postman to verify all input data

### What are the relevant pivotal tracker stories?
#163215141

### Screenshots
##### with all inputs correct
![image](https://user-images.githubusercontent.com/45405036/51166542-f5e33980-18b4-11e9-8fd1-1ac9a3c49537.png)
##### with a missing field
![image](https://user-images.githubusercontent.com/45405036/51166769-aea97880-18b5-11e9-8fad-00bcb0ee4c6d.png)
##### with an invalid email
![image](https://user-images.githubusercontent.com/45405036/51166825-d698dc00-18b5-11e9-9bb2-54bc61888a75.png)
